File: app/src/main/java/com/peakmain/basicui/activity/home/recylcer/activity/ItemTouchActivity.java
package com.peakmain.basicui.activity.home.recylcer.activity;

import android.support.v7.widget.helper.ItemTouchHelper;
import android.util.Log;

import com.peakmain.basicui.BuildConfig;
import com.peakmain.basicui.R;
import com.peakmain.basicui.activity.home.recylcer.ItemTouchHelper.GridItemTouchHelper;
import com.peakmain.basicui.activity.home.recylcer.bean.GroupBean;

/**
 * author ：Peakmain
 * createTime：2020/3/25
 * mail:2726449200@qq.com
 * describe：
 */
public class ItemTouchActivity extends BaseRecyclerAcitvity {

    @Override
    protected int getLayoutId() {
        return R.layout.basic_grid_recycler_view;
    }

    @Override
    protected void initData() {

        mRecyclerView.setAdapter(mGroupAdapter);
        GridItemTouchHelper itemTouchHelper = new GridItemTouchHelper(mGroupAdapter, mGroupBeans);
        itemTouchHelper.attachToRecyclerView(mRecyclerView);

        mGroupAdapter.adjustSpanSize(mRecyclerView);
        //获取更新后的数据
        itemTouchHelper.setOnDataUpdatedListener(datas -> {
            for (GroupBean data : datas) {
                Log.e(BuildConfig.TAG, data.isHeader ? "head" : data.getUrl());
            }
        }).setGridDragFlags(ItemTouchHelper.UP);

    }
}

File: app/src/main/java/com/peakmain/basicui/activity/home/recylcer/activity/ItemTouchActivity.kt
package com.peakmain.basicui.activity.home.recylcer.activity

import android.util.Log
import androidx.recyclerview.widget.ItemTouchHelper
import com.peakmain.basicui.BuildConfig
import com.peakmain.basicui.R
import com.peakmain.basicui.activity.home.recylcer.itemTouchHelper.GridItemTouchHelper

/**
 * author ：Peakmain
 * createTime：2020/3/25
 * mail:2726449200@qq.com
 * describe：
 */
class ItemTouchActivity : BaseRecyclerAcitvity() {
    override fun getLayoutId(): Int {
        return R.layout.basic_grid_recycler_view
    }

    override fun initData() {
        mRecyclerView!!.adapter = mGroupAdapter
        val itemTouchHelper = GridItemTouchHelper(mGroupAdapter, mGroupBeans)
        itemTouchHelper.attachToRecyclerView(mRecyclerView)
        mGroupAdapter!!.adjustSpanSize(mRecyclerView!!)
        //获取更新后的数据
        itemTouchHelper.setOnDataUpdatedListener {
            for (data in it) {
                Log.e(BuildConfig.TAG, if (data!!.isHeader) "head" else data.url)
            }
        }.setGridDragFlags(ItemTouchHelper.UP)
    }
}